### PR TITLE
Task/bws54 workers table

### PR DESCRIPTION
**Pull request by Oleg-feriss**

### CODE REVIEW

File: frontend/src/store/eam/reducer.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { EAMWorkerGetAllResponseDto } from 'common/types/types';
+
+type State = {
+  dataStatus: DataStatus;
+  workers: EAMWorkerGetAllResponseDto | null;
+};
+
+const initialState: State = {
+  dataStatus: DataStatus.IDLE,
+  workers: null,
```
Comment (what1s1ove) [inline frontend/src/store/eam/reducer.ts]:
```suggestion
  workers: EAMWorkerGetAllResponseDto; // should be EAMWorkerGetAllResponseDto.items
};

const initialState: State = {
  dataStatus: DataStatus.IDLE,
  workers: [],
```

File: shared/src/common/types/eam-worker/eam-worker-get-all-item-response-dto.type.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 type EAMWorkerGetAllItemResponseDto = {
   id: string;
   name: string;
+  createdAt: Date;
```
Comment (what1s1ove) [inline shared/src/common/types/eam-worker/eam-worker-get-all-item-response-dto.type.ts]:
```suggestion
  createdAt: string;
```

File: frontend/src/components/eam/components/workers-table/workers-table.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+const WorkersTable: FC = () => {
+  const { workers } = useAppSelector(({ eam }) => ({
+    workers: eam.workers,
+  }));
+
+  const data = useMemo(() => getRows(workers), [workers]);
+
+  const columns = useMemo(() => getColumns(), []);
+
+  return <Table columns={columns} data={data} title={'Users'} />;
```
Comment (what1s1ove) [inline frontend/src/components/eam/components/workers-table/workers-table.tsx]:
```suggestion
  return <Table columns={columns} data={data} title="Users" />;
```

File: frontend/src/components/eam/eam.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { FC } from 'react';
+import { WorkersTable } from './components/workers-table/workers-table';
+import { useAppDispatch, useEffect } from 'hooks/hooks';
+import { eam as workerActions } from 'store/actions';
```
Comment (what1s1ove) [inline frontend/src/components/eam/eam.tsx]:
```suggestion
import { eam as eamActions } from 'store/actions';
```

File: frontend/src/store/eam/common.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+enum ActionType {
+  GET_WORKERS = 'workers/',
```
Comment (what1s1ove) [inline frontend/src/store/eam/common.ts]:
```suggestion
  GET_WORKERS = 'workers/get-all',
```

File: frontend/src/components/eam/components/workers-table/helpers/get-rows.helper.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+type Row = {
+  [WorkersTableAccessor.USER_NAME]: string;
+  [WorkersTableAccessor.GROUPS]: string[];
+  [WorkersTableAccessor.CREATION_TIME]: Date;
+};
+
+const getRows = (workers: EAMWorkerGetAllResponseDto | null): Row[] => {
+  if (!workers) {
+    return [];
+  }
```
Comment (what1s1ove) [inline frontend/src/components/eam/components/workers-table/helpers/get-rows.helper.ts]:
```suggestion
```

File: frontend/src/components/eam/components/workers-table/helpers/get-rows.helper.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { WorkersTableAccessor } from 'common/enums/enums';
+import { EAMWorkerGetAllResponseDto } from 'common/types/types';
+
+type Row = {
+  [WorkersTableAccessor.USER_NAME]: string;
+  [WorkersTableAccessor.GROUPS]: string[];
+  [WorkersTableAccessor.CREATION_TIME]: Date;
+};
+
+const getRows = (workers: EAMWorkerGetAllResponseDto | null): Row[] => {
```
Comment (what1s1ove) [inline frontend/src/components/eam/components/workers-table/helpers/get-rows.helper.ts]:
```suggestion
const getRows = (workers: EAMWorkerGetAllResponseDto // should be an array of workers): Row[] => {
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by Yaroslaww-1 — DISMISSED** [review]

File: frontend/src/components/eam/eam.tsx [hunk 8]
```diff
@@ -1,4 +1,5 @@
 import { FC } from 'react';
+import { WorkersTable } from './components/workers-table/workers-table';
```
Comment (what1s1ove) [inline frontend/src/components/eam/eam.tsx]:
```suggestion
import { WorkersTable } from './components/components';
```
we already have an import

File: frontend/src/store/eam/common.ts [hunk 9]
```diff
@@ -1,4 +1,5 @@
 enum ActionType {
+  GET_WORKERS = 'workers/get-all',
```
Comment (what1s1ove) [inline frontend/src/store/eam/common.ts]:
```suggestion
  GET_WORKERS = 'eam/get-all',
```

**Comment by Oleg-feriss** [comment]
image

**Review by what1s1ove — CHANGES_REQUESTED** [review]
add space between tables 

image

as in design
image

**Review by Yaroslaww-1 — DISMISSED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**